packages/react-utilities/src/utils/assign.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Overload set for the assign(target, ...sources) function.
+ * Returns the interserction of the supplied object(s).
+ */
+export type AssignFunction = {
```

Comment:
With Variadic Type in TS we can declare this function better as: 
```ts
type UnionToIntersection<U> = 
  (U extends any ? (k: U) => void : never) extends ((k: infer I)=>void) ? I : never

type AssignFunction = <T, Sources extends any[]>(
  target: T,
  ...rest: Sources
) => T & (Sources extends (infer S)[] ? UnionToIntersection<S> : never);
```

Reply:
The definition I used is based on the type definition of `Object.assign`. I'd rather keep them both the same so that we can just replace this function with `Object.assign` if we ever remove support for older browsers.